Tests/UI/New UI/PaymentFormFactoryTests.swift

```diff
@@ -0,0 +1,47 @@
+//
+//  PaymentFormFactoryTests.swift
+//  
+//
+//  Copyright © 2022 Checkout. All rights reserved.
+//
+
+import XCTest
+import Checkout
+@testable import Frames
+
+class PaymentFormFactoryTests: XCTestCase {
+
+  func testGetPaymentFormViewController() throws {
+
+    let billingFormStyle = BillingFormFactory.defaultBillingFormStyle
+    let paymentFormStyle = BillingFormFactory.defaultPaymentFormStyle
+    let address = Address(addressLine1: "Test line1",
+                          addressLine2: nil,
+                          city: "London",
+                          state: "London",
+                          zip: "N12345",
+                          country: Country(iso3166Alpha2: "GB"))
+
+    let phone = Phone(number: "77 1234 1234",
+                      country: Country(iso3166Alpha2: "GB"))
+    let name = "User 1"
+
+    let billingForm = BillingForm(name: name, address: address, phone: phone)
+
+    let viewController = PaymentFormFactory.getPaymentFormViewController(
+      billingFormData: billingForm,
+      paymentFormStyle: paymentFormStyle,
+      billingFormStyle: billingFormStyle)
+
+    XCTAssertNotNil(viewController)
+
+    let paymentViewController = try XCTUnwrap(viewController as? PaymentViewController)
+
+    XCTAssertNotNil(paymentViewController.viewModel)
+    XCTAssertNotNil(paymentViewController.viewModel.billingFormData)
+    XCTAssertNotNil(paymentViewController.viewModel.billingFormStyle)
+    XCTAssertNotNil(paymentViewController.viewModel.paymentFormStyle)
```

Comment:
It will take a lot of time to make all objects are equatable. There are other tests to make sure all inside are tested.

Reply:
> It will take a lot of time to make all objects are equatable.
Sometimes instead of making object equatable we can just compare the properties in separate assertions. It depends how many properties there are


> There are other tests to make sure all inside are tested.
There's no unit test to confirm the data provided to the factory is the data ending up inside the returned VC or its VM. There's also no unit test to confirm that any custom changes introduced in the factory are unit tested to prevent unintended changes. There may be tests for VC and/or the VM, but we're looking at the factory, its behaviour should also be validated and constrained by appropriate unit tests imo.

Reply:
This will take alot of time to check every value in all features in frames sdk. it will cost alot of time.

Reply:
@deepesh-vasthimal-cko what do you think?

Reply:
Tried this myself, yes its not possible. Pretty disappointed as unit testing should be the goal of most design patterns and in this case this is a huge black hole we cannot enforce.